esp/src/eclwatch/XrefDetailsWidget.js

```diff
@@ -0,0 +1,177 @@
+    "dijit/Toolbar",
+    "dijit/ToolbarSeparator",
+    "dijit/TooltipDialog",
+    "dijit/TitlePane",
+    "dijit/form/TextBox",
+    "dijit/Dialog",
+    "dijit/form/SimpleTextarea",
+
+    "hpcc/TableContainer"
+], function (declare, lang, i18n, nlsHPCC, dom, domForm, domAttr, iframe, domClass, query, Memory, Observable,
+                registry,
+                OnDemandGrid, Keyboard, Selection, selector, ColumnResizer, DijitRegistry,
+                _TabContainerWidget, ESPWorkunit, ESPRequest, TargetSelectWidget, DelayLoadWidget, InfoGridWidget, WsWorkunits, GridDetailsWidget,
+                template) {
+    return declare("XrefDetailsWidget", [_TabContainerWidget], {
+        templateString: template,
+        baseClass: "XrefDetailsWidget",
+        i18n: nlsHPCC,
+
+        initalized: false,
+        loaded: false,
+        summaryWidget: null,
+        foundFilesWidget: null,
+        foundFilesWidgetLoaded: null,
+        orphanFilesWidget: null,
+        orphanFilesWidgetLoaded: null,
+        lostFilesWidget: null,
+        lostFilesWidgetLoaded: null,
+        directoriesWidget: null,
+        directoriesWidgetLoaded: null,
+        errorsWidget: null,
+        errorsWidgetLoaded: null,
+
+        postCreate: function (args) {
+            this.inherited(arguments);
+            this.summaryWidget = registry.byId(this.id + "_Summary");
+            this.foundFilesWidget = registry.byId(this.id + "_FoundFiles");
+            this.orphanFilesWidget = registry.byId(this.id + "_OrphanFiles");
+            this.lostFilesWidget = registry.byId(this.id + "_LostFiles");
+            this.directoriesWidget = registry.byId(this.id + "_Directories");
+            this.errorsWidget = registry.byId(this.id + "_Errors");
+        },
+
+        startup: function (args) {
+            this.inherited(arguments);
+        },
+
+        destroy: function (args) {
+            this.inherited(arguments);
+        },
+
+        //  Hitched actions  ---
+        _onAutoRefresh: function (event) {
+            this.wu.disableMonitor(!this.widget.AutoRefresh.get("checked"));
+        },
+        _onRefresh: function (event) {
+            this.wu.refresh(true);
+        },
+        _onGenerate: function (event) {
+
+        },
+
+        _onCancel: function (event) {
+
+        },
+
+        //  Implementation  ---
+        init: function (params) {
+            if (this.inherited(arguments))
+                return;
+
+            if (params.Name) {
+                dom.byId(this.id + "Cluster").textContent = params.Name;
+                dom.byId(this.id + "LastRun").textContent = params.Modified;
+                dom.byId(this.id + "LastMessage").textContent = params.Status;
+            }
+            this.refreshActionState();
+        },
+
+        initTab: function () {
+            var currSel = this.getSelectedChild();
+            if (currSel.id == this.widget._FoundFiles.id && !this.widget._FoundFiles.__hpcc_initalized) {
```

Comment:
I started seeing those in your code review and the `for (key` will now == `for (var key`